Addition of MEV Blocker docs to the docs. (#352)
cowprotocol/docs: docs/mevblocker/concepts/MEV Concepts/sandwich-attacks.md

---
sidebar_position: 6
---

> # What is a Sandwich Attack?

Reviewer: ```suggestion
---

# What is a Sandwich Attack?
```

> [Sandwich attacks](https://twitter.com/CoWSwap/status/1498678005777973249?ref_src=twsrc%5Etfw%7Ctwcamp%5Etweetembed%7Ctwterm%5E1498678005777973249%7Ctwgr%5Ea123df70c36d1c50e8751632901e8fe95f16372e%7Ctwcon%5Es1_&ref_url=https%3A%2F%2Fcdn.embedly.com%2Fwidgets%2Fmedia.html%3Ftype%3Dtext2Fhtmlkey%3Da19fcc184b9711e1b4764040d3dc5c07schema%3Dtwitterurl%3Dhttps3A%2F%2Ftwitter.com%2FCoWSwap%2Fstatus%2F14986780057779732493Fs3D20image%3Dhttps3A%2F%2Fi.embed.ly%2F1%2Fimage3Furl3Dhttps253A252F252Fabs.twimg.com252Ferrors252Flogo46x38.png26key3Da19fcc184b9711e1b4764040d3dc5c07) occur when a user's transaction gets trapped, or "sandwiched," between two hostile transactions — one before and one after. As a result, the original transaction executes at a much higher price than necessary, leading to an inflated price for the original trader and a profit for the malicious trader placing the two extra trades (known as a "searcher").

Reviewer: ```suggestion
[Sandwich attacks](https://twitter.com/CoWSwap/status/1498678005777973249?ref_src=twsrc%5Etfw%7Ctwcamp%5Etweetembed%7Ctwterm%5E1498678005777973249%7Ctwgr%5Ea123df70c36d1c50e8751632901e8fe95f16372e%7Ctwcon%5Es1_&ref_url=https%3A%2F%2Fcdn.embedly.com%2Fwidgets%2Fmedia.html%3Ftype%3Dtext2Fhtmlkey%3Da19fcc184b9711e1b4764040d3dc5c07schema%3Dtwitterurl%3Dhttps3A%2F%2Ftwitter.com%2FCoWSwap%2Fstatus%2F14986780057779732493Fs3D20image%3Dhttps3A%2F%2Fi.embed.ly%2F1%2Fimage3Furl3Dhttps253A252F252Fabs.twimg.com252Ferrors252Flogo46x38.png26key3Da19fcc184b9711e1b4764040d3dc5c07) occur when a user's transaction gets trapped, or "sandwiched," between two hostile transactions - one before and one after. As a result, the original transaction executes at a much higher price than necessary, leading to an inflated price for the original trader and a profit for the malicious trader placing the two extra trades (known as a "searcher").
```

Hyphen, not em-dash (source code requirement).

* **Price Impact:** Crypto markets, like all markets, are based on supply and demand. AMMs maintain "liquidity pools" of assets that they use to fill trades. Each trade drains some amount of this liquidity, moving the price of the asset. This price movement is known as "price impact." The larger the trade, the bigger the price impact. A trade of $100 ETH will not move the price of ETH very noticeably, since it makes up a tiny fraction of the available liquidity. A trade of $1,000,000 ETH however, will noticeably move the price of ETH.

> * **Transaction Reordering:** Blockchain transactions do not always enter the block in the order that they were submitted. Searchers can "bribe" the validators responsible for creating the block to get them to arrange transactions in a specific sequence. This transaction reordering is what makes all of MEV, including sandwich attacks, possible.

Reviewer: ```suggestion
* **Transaction Reordering:** Blockchain transactions do not always enter the block in the order that they were submitted. Searchers can "bribe" the validator responsible for creating the block to have them arrange transactions in a specific sequence. This transaction reordering is what makes all of MEV, including sandwich attacks, possible.
```

## The sandwich attack: Step by step

Let's examine a sandwich attack through a step-by-step example. In this example, we're trading ETH and COW.

> ![Sandiwch](/img/mevblocker/Sandwich_Attack.webp)

Reviewer: ```suggestion
![Sandwich](/img/mevblocker/Sandwich_Attack.webp)
```

* **Step 1:** Bessie wants to buy COW using her ETH. She goes to a decentralized exchange (DEX) like Uniswap, and places an order for 4,000 COW. This should cost her around 1 ETH, but due to significant market volatility, Bessie decides to set a 10% slippage tolerance. This means she's willing to pay up to 1.1 ETH for 4,000 COW.

* **Step 2:** Bessie's trade enters the Ethereum mempool (the pending order queue), and a lurking searcher spots an opportunity. Springing into action, the searcher places a trade just before Bessie's large enough to push the COW price up to her slippage tolerance. In this case, the searcher buys 4,000 COW for exactly 1 ETH. As a result of this first trade's price impact, Bessie's 4,000 COW purchase now costs 1.1 ETH — the maximum she's willing to pay.

> * **Step 3:**a Once the searcher's transaction clears, Bessie's transaction also goes through and she receives her 4,000 COW in exchange for 1.1 ETH. The searcher takes advantage of this price impact and sells his original COW at this new rate — 4,000 COW for 1.1 ETH. In the end, the searcher buys 4,000 COW for 1 ETH and sells it for 1.1 ETH, earning a profit of 0.1 ETH (before gas and fees) for not much effort.

Reviewer: ```suggestion
1. Bessie wants to buy COW using her ETH. She goes to a decentralized exchange (DEX) like Uniswap, and places an order for 4,000 COW. This should cost her around 1 ETH, but due to significant market volatility, Bessie decides to set a 10% slippage tolerance. This means she's willing to pay up to 1.1 ETH for 4,000 COW.
2. Bessie's trade enters the Ethereum mempool (the pending order queue), and a lurking searcher spots an opportunity. Springing into action, the searcher places a trade just before Bessie's large enough to push the COW price up to her slippage tolerance. In this case, the searcher buys 4,000 COW for exactly 1 ETH. As a result of this first trade's price impact, Bessie's 4,000 COW purchase now costs 1.1 ETH - the maximum she's willing to pay.
3. Once the searcher's transaction clears, Bessie's transaction also goes through and she receives her 4,000 COW in exchange for 1.1 ETH. The searcher takes advantage of this price impact and sells their original COW at this new rate - 4,000 COW for 1.1 ETH. In the end, the searcher buys 4,000 COW for 1 ETH and sells it for 1.1 ETH, earning a profit of 0.1 ETH (before gas and fees) for not much effort.
```

Don't use numbers in bullet points. Just use a numbered list.

Author: Sure, this is just how we had it in the article, so should be fine with that format.
